core/src/platform.cpp

```diff
@@ -46,4 +47,92 @@ std::vector<FontSourceHandle> Platform::systemFontFallbacksHandle() const {
     return {};
 }
 
+void Platform::shutdown() {
+    if (m_shutdown.exchange(true)) { return; }
+
+    {
+        std::lock_guard<std::mutex> lock(m_callbackMutex);
+
+        UrlResponse response;
+        response.error = "Shutting down";
+
+        for (auto& entry : m_urlCallbacks) {
+            auto& request = entry.second;
+            if (request.callback) {
+                request.callback(std::move(response));
+            }
+            if (request.id) {
+                urlRequestCanceled(request.id);
+            }
+        }
+        m_urlCallbacks.clear();
+    }
+}
+
+UrlRequestHandle Platform::startUrlRequest(Url _url, UrlCallback&& _callback) {
+    assert(_callback);
+
+    if (m_shutdown) {
+        UrlResponse response;
+        response.error = "Shutting down";
+        if (_callback) {
+            _callback(std::move(response));
+        }
+        return 0;
+    }
+
+    UrlRequestHandle handle= ++m_urlRequestCount;
+
+    // Need to do this in advance in case startUrlRequest calls back synchronously..
+    UrlRequestEntry* entry = nullptr;
+    {
+        std::lock_guard<std::mutex> lock(m_callbackMutex);
+        auto it = m_urlCallbacks.emplace(handle, UrlRequestEntry{std::move(_callback), 0});
```

Comment:
Instead of using `0`, use the `constexpr` defined above: `UrlRequestNotCancelable`